tsdb/querier.go

```diff
@@ -855,13 +855,18 @@ func (p *populateWithDelChunkSeriesIterator) Next() bool {
 		if app, err = newChunk.Appender(); err != nil {
 			break
 		}
-
-		for vt := valueType; vt != chunkenc.ValNone; vt = p.currDelIter.Next() {
+		var h *histogram.Histogram
```

Comment:
Perhaps leave a short code comment here stating that we need to set the MinTime to the first sample in the chunk, which might be different from before. (And same below.) So that future readers don't wonder too much about this prelude to the for loop below.

Reply:
I've just realized that you are just doing what has been done for float samples all the time. (And we just forgot to do it when doing the histogram code.) So this is just normal and probably does not need a comment. (Or otherwise, we would need to add it to the float case, too. Your call…)

Reply:
Okay, no comment then (: